src/java.desktop/macosx/classes/com/apple/laf/AquaMenuPainter.java

```diff
@@ -410,7 +408,15 @@ protected void paintArrow(Graphics g, JMenuItem c, ButtonModel model, Icon arrow
         if (c instanceof JMenu && (model.isArmed() || model.isSelected()) && arrowIcon instanceof InvertableIcon) {
             ((InvertableIcon)arrowIcon).getInvertedIcon().paintIcon(c, g, arrowIconRect.x, arrowIconRect.y);
         } else {
-            arrowIcon.paintIcon(c, g, arrowIconRect.x, arrowIconRect.y);
+            if (!c.isEnabled() && arrowIcon instanceof ImageIcon) {
+                if (disabledIcon == null) {
+                    disabledIcon = new ImageIconUIResource(GrayFilter.
+                            createDisabledImage(((ImageIcon) arrowIcon).getImage()));
+                }
+                disabledIcon.paintIcon(c, g, arrowIconRect.x, arrowIconRect.y);
+            } else {
+                arrowIcon.paintIcon(c, g, arrowIconRect.x, arrowIconRect.y);
+            }
```

Comment:
It will be even better to implement it in the same way as done in other L&fs like windows/metal.

Reply:
I will like to know how because as it is pointed out, paintArrow delegates drawing to ImageIcon#paintIcon which is in shared code and this is mac specific issue so it needs to be handled before we call ImageIcon#paintIcon

Reply:
You need to override the paintIcon method in the InvertableImageIcon returned by the AquaImageFactory.getMenuArrowIcon() or you can create `class MenuArrowIcon extends InvertableImageIcon,` and override it there.

Also take a look to another usage of InvertableImageIcon for "MenuItemCheckIcon", should we disable it as well?

Reply:
MenuItemCheckIcon disable-ness is also solved along with this fix.
JCheckBoxMenuItem component for MenuItemCheckIcon test is added additionally.

Reply:
Any more comments on this?